### ISSUE TITLE
Exp orbs causing other particles to rotate

### ISSUE THREAD

**Issue by xanthousm** [ferriarnus/VivecraftForge#55]
When an exp orb is rendered, its orientation is applied to all other particles. Screenshots below are from testing on fabric without any other mods installed. The first shows the exp orb (bottom left) that is affecting the smoke particle's orientation, the second shows how smoke and potion particles are being rotated by an exp orb directly below the player.

This bug is in the latest versions of both the 1.18 and 1.19 branches.

When iris+sodium are installed and a shader is used, this bug only appears in the left eye.

![2022-10-06_10 54 33](https://user-images.githubusercontent.com/70198941/194185005-356a3eaa-d4c0-427e-83a1-d025b56a836a.png)
![Untitled](https://user-images.githubusercontent.com/70198941/194185020-44baf674-7bc6-48af-8420-11c6a410618b.png)

**Comment by ghost** [comment]
likely an issue with billboarding textures

**Comment by xanthousm** [comment]
I found the issue. The variable `cameraorientation` (used to rotate all particles) is being modified in two methods in `mixin\client\renderer\entity\EntityRenderDispatcherVRMixin.java`, when the methods should just be assigning a return value for each particle-like entity (snowballs and entity nametags caused the same issue). 

As a fix, I just used a local Quaternion in the return statements instead (ec706f8367b3f47114be629637241fcb7f1d7259).
